portfolio/src/main/java/com/google/sps/servlets/PostServlet.java

```diff
@@ -0,0 +1,116 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.SortDirection;	
+import com.google.gson.Gson;
+import com.google.sps.data.BlogPost;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/post")
+public class PostServlet extends HttpServlet {
+
+  /** The post param is a string property identifier for setting and retrieving
+   * the body of textual content stored for a blog post made by the user. */
+  private static final String POST_PARAM = "blogpost";
+
+  /** The postin param is a unique string identifier for retrieving the blog 
+   * post text from a user through the form input in POST requests */
+  private static final String POSTIN_PARAM = "blog-input";
+  
+  /** The title param is string property identifier for setting and retrieving
+   * the title of the blog post created. If not included in a GET request, no 
+   * blog title will be returned. */
+  private static final String TITLE_PARAM = "title";
+
+  /** The time param is a string property identifier for the time of submission 
+   * upon a POST request for a submitted blog post. It is also the property 
+   * used to sort posts in most recently submitted order */
+  private static final String TIME_PARAM = "timestamp";
+
+  /** The category pram is a string property identifier for the category of a 
+   * blog post that will be retrieved via GET requests for filtering post by
+   * category type. Many posts may have the same category */
+  private static final String CAT_PARAM = "category";
+
+  /** The tag param is a string property identifier for blog posts. It is the 
+   * property used to determine which blog posts are displayed with their 
+   * corresponding blog posts. If not included in a GET request, no blog posts 
```

Comment:
Doesn't seem accurate - below, if TAG_PARAM is not included, blog posts are still returned.